src/main/java/org/truffleruby/language/SafepointManager.java

```diff
@@ -204,8 +205,11 @@ private void restoreDefaultInterruptHandler() {
         Log.LOGGER.warning("restoring default interrupt handler");
 
         try {
-            final Signal signal = context.getNativePlatform().getSignalManager().createSignal("INT");
-            context.getNativePlatform().getSignalManager().watchDefaultForSignal(signal);
+            final Signal signal = new Signal("INT");
+            final sun.misc.SignalHandler defaultHandler = Signals.DEFAULT_HANDLERS.get(signal);
+            if (defaultHandler != null) { // otherwise it is already the default signal
+                Signal.handle(signal, defaultHandler);
+            }
```

Comment:
This was better encapsulated before, particularly Signals.DEFAULT_HANDLERS which is now accessed as global public mutable state.
Could we have `Signals.watchDefaultForSignal()` and `registerSignal()` so `DEFAULT_HANDLERS` is the implementation secret of Signals and doesn't need to leak in all usages?

Reply:
#785